homeassistant/components/permobil/sensor.py

```diff
@@ -53,6 +55,18 @@ class PermobilSensorEntityDescription(
     """Describes Permobil sensor entity."""
 
 
+def records_dist_fn(data: Any) -> float:
+    """Find and calculate the record distance.
+
+    This function gets the record distance from the data and converts it to kilometers if needed.
+    """
+    distance = data.records[RECORDS_DISTANCE[0]]
+    distance_unit = data.records[RECORDS_DISTANCE_UNIT[0]]
+    if distance_unit != KM:
+        distance = distance * MILES_TO_KM
+    return distance
```

Comment:
Wait, instead of transforming it to KM, can we maybe set the native UoM to the unit of the record?

Reply:
I did is this way as i deemed it easier and since none of the other sensors have a dynamic UoM. 
If i were to change this approach should i then also make a new Callable for each existing sensor description?

Reply:
```py

    @property
    def native_unit_of_measurement(self) -> str:
        """Return the unit of measurement of the sensor."""
        if self.entity_description.key == "record_distance":
            return {
                KM: UnitOfLength.KILOMETERS,
                MILES: UnitOfLength.MILES,
            }.get(self.coordinator.data.records[RECORDS_DISTANCE_UNIT[0]])
        return self.entity_description.native_unit_of_measurement
```
What do you think of this? You could extract that dict out of the function, but now you get the full picture

Reply:
I think that's a very good solution thank you! I will do it that way.